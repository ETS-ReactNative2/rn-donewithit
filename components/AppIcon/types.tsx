import { StyleProp, TextStyle } from 'react-native'

export type AppIconType = {
  name?: any
  size?: number
  color?: string
  style?: StyleProp<TextStyle>
}
