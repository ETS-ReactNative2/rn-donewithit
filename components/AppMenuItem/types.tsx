import { ReactNode } from 'react'
import { GestureResponderEvent, StyleProp, ViewStyle } from 'react-native'

export type AppMenuItemProps = {
  iconName?: any
  iconBackgroundColor: string
  title: string
  subTitle?: string
  style?: StyleProp<ViewStyle>
  onPress?: (e?: GestureResponderEvent) => void
  IconComponent?: ReactNode
}
