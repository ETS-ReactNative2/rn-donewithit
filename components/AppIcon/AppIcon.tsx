import React from 'react'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import { AppIconType } from './types'
import CONSTS from './consts'

function AppIcon({
  name = CONSTS.NAME,
  size = CONSTS.SIZE,
  color = CONSTS.COLOR,
  style
}: AppIconType) {
  return (
    <MaterialCommunityIcons
      style={style}
      name={name}
      size={size}
      color={color}
    />
  )
}

export default AppIcon
