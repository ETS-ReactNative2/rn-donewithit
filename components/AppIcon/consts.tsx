import COLORS from '../../config/colors'

const CONSTS = {
  NAME: 'email',
  SIZE: 20,
  COLOR: COLORS.WHITE
}

export default CONSTS
