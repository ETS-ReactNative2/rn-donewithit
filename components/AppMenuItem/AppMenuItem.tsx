import React from 'react'
import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import COLORS from '../../config/colors'
import CONSTS from './consts'
import AppText from '../AppText/AppText'

type AppMenuItemProps = {
  iconName: any
  iconBackgroundColor: string
  title: string
  style?: StyleProp<ViewStyle>
}
function AppMenuItem({
  iconName,
  iconBackgroundColor,
  title,
  style
}: AppMenuItemProps) {
  return (
    <View style={[style, styles.menu]}>
      <View style={[styles.menuIcon, { backgroundColor: iconBackgroundColor }]}>
        <MaterialCommunityIcons
          name={iconName}
          size={CONSTS.ICON_SIZE}
          color={COLORS.WHITE}
        />
      </View>
      <AppText style={styles.menuTitle}>{title}</AppText>
    </View>
  )
}

const styles = StyleSheet.create({
  menu: {
    padding: CONSTS.PADDING,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: COLORS.WHITE
  },
  menuIcon: {
    width: CONSTS.ICON_WRAP_SIZE,
    height: CONSTS.ICON_WRAP_SIZE,
    borderRadius: CONSTS.ICON_WRAP_SIZE / 2,
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: CONSTS.PADDING / 2
  },
  menuTitle: {
    fontSize: CONSTS.FONT_SIZE,
    textTransform: 'capitalize'
  }
})

export default AppMenuItem
