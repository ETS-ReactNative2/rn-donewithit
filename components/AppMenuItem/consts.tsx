const CONSTS = {
  PADDING: 20,
  ICON_WRAP_SIZE: 40,
  ICON_SIZE: 20,
  FONT_SIZE: 18
}

export default CONSTS
