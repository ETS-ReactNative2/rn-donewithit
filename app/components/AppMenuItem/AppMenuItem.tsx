import React, { useCallback } from 'react'
import {
  View,
  StyleSheet,
  GestureResponderEvent,
  TouchableHighlight
} from 'react-native'
import COLORS from '../../config/colors'
import CONSTS from './consts'
import AppText from '../AppText/AppText'
import { AppMenuItemProps } from './types'
import AppIcon from '../AppIcon'

function AppMenuItem({
  iconName,
  iconBackgroundColor,
  title,
  subTitle,
  style,
  onPress,
  IconComponent
}: AppMenuItemProps) {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableHighlight
      underlayColor={COLORS.LIGHT_GRAY}
      activeOpacity={0.85}
      onPress={(e?: GestureResponderEvent) => {
        handlePress(e)
      }}
    >
      <View style={[style, styles.menu]}>
        <View
          style={[styles.menuIcon, { backgroundColor: iconBackgroundColor }]}
        >
          {IconComponent || (
            <AppIcon
              name={iconName}
              size={CONSTS.ICON_SIZE}
              color={COLORS.WHITE}
            />
          )}
        </View>
        <View style={styles.infoBox}>
          <AppText style={styles.menuTitle}>{title}</AppText>
          {subTitle && (
            <AppText style={[styles.menuTitle, styles.menuSubTitle]}>
              {subTitle}
            </AppText>
          )}
        </View>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  menu: {
    padding: CONSTS.PADDING,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: COLORS.WHITE
  },
  menuIcon: {
    width: CONSTS.ICON_WRAP_SIZE,
    height: CONSTS.ICON_WRAP_SIZE,
    borderRadius: CONSTS.ICON_WRAP_SIZE / 2,
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: CONSTS.PADDING / 2
  },
  infoBox: {
    flexDirection: 'column'
  },
  menuTitle: {
    fontSize: CONSTS.FONT_SIZE,
    textTransform: 'capitalize'
  },
  menuSubTitle: {
    color: COLORS.GRAY,
    marginTop: 5
  }
})

export default AppMenuItem
