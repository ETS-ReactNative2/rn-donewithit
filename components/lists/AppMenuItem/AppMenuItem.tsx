import React, { useCallback } from 'react'
import { View, GestureResponderEvent, TouchableHighlight } from 'react-native'
import COLORS from '../../../config/colors'
import CONSTS from './consts'
import styles from './styles'
import AppText from '../../AppText/AppText'
import { AppMenuItemProps } from './types'
import AppIcon from '../../AppIcon'

function AppMenuItem({
  iconName,
  iconBackgroundColor,
  title,
  subTitle,
  style,
  onPress,
  IconComponent
}: AppMenuItemProps) {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableHighlight
      underlayColor={COLORS.LIGHT_GRAY}
      activeOpacity={0.85}
      onPress={(e?: GestureResponderEvent) => {
        handlePress(e)
      }}
    >
      <View style={[style, styles.menu]}>
        <View
          style={[styles.menuIcon, { backgroundColor: iconBackgroundColor }]}
        >
          {IconComponent || (
            <AppIcon
              name={iconName}
              size={CONSTS.ICON_SIZE}
              color={COLORS.WHITE}
            />
          )}
        </View>
        <View style={styles.infoBox}>
          <AppText style={styles.menuTitle}>{title}</AppText>
          {subTitle && (
            <AppText style={[styles.menuTitle, styles.menuSubTitle]}>
              {subTitle}
            </AppText>
          )}
        </View>
      </View>
    </TouchableHighlight>
  )
}

export default AppMenuItem
