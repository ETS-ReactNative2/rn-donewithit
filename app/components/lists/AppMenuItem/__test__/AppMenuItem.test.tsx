import React from 'react'
import renderer from 'react-test-renderer'
import AppIcon from '../../../AppIcon'
import AppMenuItem from '../index'
import COLORS from '../../../../config/colors'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const menu = {
  iconName: 'format-list-bulleted',
  iconBackgroundColor: COLORS.PRIMARY,
  title: 'my listings'
}

describe('AppMenuItem', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <AppMenuItem
          title={menu.title}
          IconComponent={<AppIcon name={menu.iconName} />}
          iconBackgroundColor={menu.iconBackgroundColor}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
