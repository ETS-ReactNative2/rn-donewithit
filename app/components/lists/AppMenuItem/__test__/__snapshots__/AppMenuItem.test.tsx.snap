// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenuItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "padding": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 20,
            "height": 40,
            "justifyContent": "center",
            "marginRight": 10,
            "overflow": "hidden",
            "width": 40,
          },
          Object {
            "backgroundColor": "tomato",
          },
        ]
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Courier",
              "fontSize": 13,
            },
            Object {
              "fontSize": 18,
              "textTransform": "capitalize",
            },
          ]
        }
      >
        my listings
      </Text>
    </View>
  </View>
</View>
`;
